source/fundamentals/connection/socks5.txt

```diff
@@ -0,0 +1,113 @@
+.. _kotlin-connect-socks:
+
+==========================================
+Connect to MongoDB by Using a SOCKS5 Proxy
+==========================================
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code example, security, connection string
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+driver-long+} to connect
+to MongoDB by using a **SOCKS5 proxy**. SOCKS5 is a standardized
+protocol for communicating with network services through a proxy server.
+
+.. tip::
+
+   To learn more about the SOCKS5 protocol, see the Wikipedia entry on
+   :wikipedia:`SOCKS <w/index.php?title=SOCKS&oldid=1160087146>`.
+
+.. _kotlin-socks-proxy-settings:
+
+SOCKS5 Proxy Settings
+---------------------
+
+The proxy settings specify the SOCKS5 proxy server address and your
+authentication credentials. You can specify your settings in an instance of
+``MongoClientSettings`` or in your connection string.
+
+The following table describes the SOCKS5 client options:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 15 20 65
+
+   * - Name
+     - Accepted Values
+     - Description
+
+   * - **proxyHost**
+     - String
+     - Specifies the SOCKS5 proxy IPv4 address, IPv6 address, or hostname.
+       You must provide this value to connect to a SOCKS5 proxy.
+
+   * - **proxyPort**
+     - non-negative Integer
+     - Specifies the TCP port number of the SOCKS5 proxy server. This option
+       defaults to ``1080`` when you set ``proxyHost``.
```

Comment:
I'm a bit confused by the last sentence here. It says it defaults to 1080 "when you set proxyHost" but from the description of proxyHost it seems like `proxyHost` is required? So does that mean this will always default to 1080? 
I think my confusion is that this sentence also implies that there is another default, but doesn't specify it. Could you add some clarification here, or possibly remove the "when you set..." part if that still makes sense

Reply:
I think the way it works is that if you set `proxyHost` but do not set `proxyPort`, `proxyPort` defaults to 1080. However, you can set `proxyPort` manually to whatever you want. If you do not set `proxyHost`, then there is no default for `proxyPort` and it also remains unset. I can clear up that wording.